### ISSUE TITLE
Get Started flow broken when JavaScript turned off

### ISSUE THREAD

**Issue by zoechi** [flutter/website#1853]
@filiph commented on [Tue Feb 27 2018](https://github.com/flutter/flutter/issues/14922)

From URL: https://docs.flutter.dev/get-started/editor?tab=vscode

The new startup flow won't let you choose an alternative when your JavaScript is turned off. JavaScript is turned off for about 2% of browsers (not sure what the stats are for developers, I would say higher than average).

If we choose to act on this, one solution is to show both versions by default and let JavaScript hide the ones that aren't applicable on page load. Another solution is to show the rest of the options with `<noscript>`. It can be very ugly, just show the content.

**Comment by chalin** [comment]
Is this still relevant @filiph? I don't think that the new site is usable w/o JS enabled. /cc @sfshaza2 @kwalrath

**Comment by filiph** [comment]
This is not P0, but I think we should at least attempt to show content for no-JS users. If the new site isn't usable at all without JS, then I think that's a problem we should raise with the developers of the revamp.

Again, the no-JS version can be ugly and sub-par. But at least people should be able to access the text.

**Comment by chalin** [comment]
Actually, we're in pretty good shape. I'd say that over 90% of the site content is accessible and readable. What remains inaccessible is the following:

- Pages with tabbed content, such as the **Get started** [Set up an editor](https://flutter.io/docs/get-started/editor) and [Test drive](https://flutter.io/docs/get-started/test-drive) pages. For such pages, the user sees the content of the default tab (which is Android Studio / IntelliJ for these two pages in particular).
- **Sidenav** open/close functionality for top-level entries doesn't work. Thankfully, some top-level entries are open by default, and for those, clicking on a subentry directs the user to the appropriate page. I can probably update the sidenav so that all entrys would be functional. Let me know if you'd like for me to do so.
- On mobile, the hamburger doesn't work. So users need to visit pages directly, or use other navigation mechanisms (for pages that have them), like breadcrumbs or next/prev page links.
- Search doesn't work.

Is this good enough for a ~2% of users? If not, let me know which specific points that you'd like for me to address.

**Comment by filiph** [comment]
Ok, I think this is good enough, to be honest. It seems that, when a no-JS-by-default user clicks on a link going to our site, they'll be able to read the content. Then they can either allow JS for our domain, or continue using the site in a limited fashion.

Is it clear for no-JS users that the stuff doesn't work because of no-JS? Maybe it might make sense to add a non-intrusive but persistent `<noscript>` banner that says something like: "It seems that your browser doesn't support (or allow) JavaScript. This site uses JavaScript for some of its functionality, like search, menu drill-down, and more.".

**Comment by chalin** [comment]
SG

**Comment by atsansone** [comment]
With the new flow for install, most of the page is usable. Closing.
